### PR TITLE
use overlay2 for janky and experimental checks

### PR DESCRIPTION
**Pull request by andrewhsu**
to address #39561

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -74,7 +74,7 @@ pipeline {
                   -v "$WORKSPACE/.git:/go/src/github.com/docker/docker/.git" \
                   --name docker-pr$BUILD_NUMBER \
                   -e DOCKER_GITCOMMIT=${GITCOMMIT} \
-                  -e DOCKER_GRAPHDRIVER=vfs \
+                  -e DOCKER_GRAPHDRIVER=overlay2 \
                   -e DOCKER_EXECDRIVER=native \
```
Comment (AkihiroSuda) [inline Jenkinsfile]: not related to this PR, but probably we can remove DOCKER_EXECDRIVER

Comment (thaJeztah) [reply Jenkinsfile]: Yes, we should remove it

File: Jenkinsfile [hunk 2]
```diff
@@ -125,7 +125,7 @@ pipeline {
                     -e DOCKER_EXPERIMENTAL=y \
                     --name docker-pr-exp$BUILD_NUMBER \
                     -e DOCKER_GITCOMMIT=${GITCOMMIT} \
-                    -e DOCKER_GRAPHDRIVER=vfs \
+                    -e DOCKER_GRAPHDRIVER=overlay2 \
```
Comment (thaJeztah) [inline Jenkinsfile]: Would it work if we _removed_ `=overlay2` ? Then we can set `DOCKER_GRAPHDRIVER` in jenkins-configuration if we want to change it (thinking out loud)

Comment (andrewhsu) [reply Jenkinsfile]: for now, trying to keep number of knobs to turn to a minimum. perhaps in the future, but wants to get the jenkinsfile working and dependable first.

File: Jenkinsfile (hunk 1 above)
Comment (andrewhsu) [reply Jenkinsfile]: #39606

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by cpuguy83** [comment]
Conflicts now :(

**Comment by andrewhsu** [comment]
@cpuguy83 fyi i've resolved conflicts

**Pull request merged**